lib/flop/builder.ex

```diff
@@ -30,20 +30,27 @@ defmodule Flop.Builder do
     :ilike_or
   ]
 
-  def filter(_, %Filter{field: nil}, c), do: c
-  def filter(_, %Filter{value: nil}, c), do: c
-
-  def filter(
-        schema_struct,
-        %Filter{field: field} = filter,
-        conditions
-      ) do
-    build_op(
-      conditions,
-      schema_struct,
-      get_field_type(schema_struct, field),
-      filter
-    )
+  def filter(query, schema_struct, filters) do
+    {query, conditions} =
+      filters
+      |> Enum.reduce({query, true}, fn
+        %Filter{field: nil}, acc ->
+          acc
+
+        %Filter{value: nil}, acc ->
+          acc
+
+        %Filter{field: field} = filter, {q, c} ->
+          case get_field_type(schema_struct, field) do
+            {:custom, {mod, fun, opts}} ->
+              {apply(mod, fun, [q, filter, opts]), c}
```

Comment:
I also wonder if it would make sense to enable passing options through that are passed to the flop functions, if the filter require extra request context to work (like allowed filters for certain users, or the timezone of the user, etc)

Wdyt?

Reply:
Might be useful. We could add another `extensions` or `extra` option key for that to avoid naming collisions. The value should be a keyword list. Not sure about the naming - are there examples from other Elixir libraries we can follow? `extensions` comes from the Graphql world.

Reply:
Something with `_opts` in the name feels more like what I have seen so far, since then it is clear that the value is a keyword list. So `extra_opts` or `custom_opts` maybe?

Reply:
Went for `extra_opts` for now.